### PR TITLE
Reduce CI Time by separating Prophet tests

### PR DESCRIPTION
**Pull request by ParthivNaresh**
Fixes: #2588 

This will bring `git-test-other` down from 19/20 min to 5/6 min. In anticipation of integrating Prophet into AutoML, running `git-test-automl` means that many of our tests that rely on instantiating every component/estimator will still need to instantiate Prophet, so I don't think removing it from `git-test-automl` is a good idea.

### CODE REVIEW

File: docs/source/release_notes.rst [hunk 1]
```diff
@@ -8,6 +8,7 @@ Release Notes
         * Deleted ``_put_into_original_order`` helper function :pr:`2639`
     * Documentation Changes
     * Testing Changes
+        * Speed up CI time by splitting Prophet tests into a separate workflow in GitHUB :pr:`2644`
```
Comment (gsheni) [inline docs/source/release_notes.rst:11]:
```suggestion
        * Speed up CI time by splitting Prophet tests into a separate workflow in GitHub :pr:`2644`
```

File: Makefile [hunk 2]
```diff
@@ -59,7 +59,11 @@ git-test-other-core:
...
+	pytest evalml/tests --ignore evalml/tests/automl_tests/ --ignore evalml/tests/tuner_tests/ --ignore evalml/tests/model_understanding_tests/ --ignore evalml/tests/pipeline_tests/ --ignore evalml/tests/objective_tests/ --ignore evalml/tests/utils_tests/ --ignore evalml/tests/component_tests/test_prophet_regressor.py --ignore evalml/tests/component_tests/test_components.py --ignore evalml/tests/component_tests/test_utils.py -n 2 --durations 0 --timeout 300 --doctest-modules --cov=evalml --junitxml=test-reports/git-test-other-junit.xml --doctest-continue-on-failure
+
+.PHONY: git-test-prophet
+git-test-prophet:
+	pytest evalml/tests/component_tests/test_prophet_regressor.py evalml/tests/component_tests/test_components.py evalml/tests/component_tests/test_utils.py evalml/tests/pipeline_tests/ evalml/tests/objective_tests/ evalml/tests/utils_tests/ -n 2 --durations 0 --timeout 300 --doctest-modules --cov=evalml --junitxml=test-reports/git-test-prophet-junit.xml --doctest-continue-on-failure
```
Comment (ParthivNaresh) [inline Makefile:66]: Due to how many of our tests are constructed, we end up instantiating every estimator including Prophet, so I had to break up `git-test-other` to focus on the files that end up requiring `Prophet`. This is also the reason that I'm leaving `git-test-automl` as it is.

**Review by chukarsten — APPROVED** [review]
This looks good to merge to me

**Review by bchen1116 — APPROVED** [review]
LGTM!

File: Makefile (hunk 2 above)
Comment (freddyaboulton) [reply Makefile:66]: Shouldn't the `objective_tests` be in `test-other`? They don't depend on prophet right?

**Review by freddyaboulton — APPROVED** [review]
Thank you @ParthivNaresh !

**Pull request merged**